ResourceLeak.java

```diff
@@ -0,0 +1,66 @@
+package com.mariten.kanatools;
+
+import com.sun.org.slf4j.internal.LoggerFactory;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class ResourceLeak {
+	private boolean active;
+    public void setActive(boolean b)
+          {
+            this.active = b;
+          }
+
+    LoggerFactory logger = new LoggerFactory();
+	String count;
+    public void process() {
+        SecureRandom secureRandom = new SecureRandom();
+	     System.runFinalizersOnExit(true);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Dm: Method invokes dangerous method runFinalizersOnExit (DM_RUN_FINALIZERS_ON_EXIT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dm-method-invokes-dangerous-method-runfinalizersonexit-dm-run-finalizers-on-exit).
*Never call System.runFinalizersOnExit or Runtime.runFinalizersOnExit for any reason: they are among the most dangerous methods in the Java libraries.* -- Joshua Bloch